proposals/lip-0053.md

```diff
@@ -125,9 +125,10 @@ def verifyLivenessConditionForRegisteredChains(ccu: Transaction) -> None:
 #### verifyCertificate
 
 ```python
-def verifyCertificate(ccu: Transaction) -> None:
+def verifyCertificate(ccu: CCU) -> None:
     certificateBytes = ccu.params.certificate      
     certificate = decode(certificateSchema, certificateBytes)
+    validateObjectSchema(certificateSchema, certificate)
```

Comment:
@ricott1 Is this implemented like this already? In principle, this is a protocol change here.

Reply:
It looks to me that the schema check is not done here: https://github.com/LiskHQ/lisk-sdk/blob/bf04175f212550492b0b360972ddf1a4d5dabf31/framework/src/modules/interoperability/base_interoperability_internal_methods.ts#L378

Notice however that in principle we should already check in `verifyLivenessConditionForRegisteredChains` (which is also not done), or maybe move the certificate verification to the top, although maybe it would not be so nice in the SDK implementation, as it is only done in the mainchain. @shuse2 can maybe pitch in here as well.

Reply:
I guess it is not important in which order it is done. If the serialized certificate cannot be decoded, then `verifyLivenessConditionForRegisteredChains` will throw. If it can be decoded but the schema is obeyed, then `verifyCertificate` does throw.

Reply:
I would still find it nicer to have `verifyCertificate(ccu)` before `verifyLivenessConditionForRegisteredChains(ccu)` as it is not good practice to work with an input object that is not fully validated and only do a validation later.

I would, however, not want to make this change as part of this PR as it is out-of-scope.

Therefore, my suggestion would be that you open any issue that you see necessary for Interoperability and we could resolve this discussion here.

Reply:
The only problem is that in the implementation the common verification is done together, so it would be a bit awkward to interpose the verification for liveness done only on mainchain. I can create the issue in the meanwhile

Reply:
can be resolved

Reply:
As discussed, @ricott1 will still check this point.